node_pools.tf

```diff
@@ -258,11 +259,180 @@ resource "google_container_node_pool" "node_pool_dynamic_pods" {
+
+    # COS_CONTAINERD is required for sandbox_config to work
+    image_type = var.enable_gvisor_dynamic_blue ? "COS_CONTAINERD" : "COS"
+
+    # Only include sandbox config if we are using gvisor
+    dynamic "sandbox_config" {
+      for_each = var.enable_gvisor_dynamic_blue ? ["placeholder"] : []
+      content {
+        sandbox_type = "gvisor"
+      }
+    }
+
+  }
+}
+
+## Green dynamic pods pool (added 2020-12-16)
+
+resource "google_container_node_pool" "dynamic_green_node_pool" {
+  count = var.enable_dynamic_green_node_pool ? 1 : 0
+
+  provider = google-beta
+
+  # version    = data.google_container_cluster.primary.master_version
+
+  # this one can take a long time to delete or create
+  timeouts {
+    create = "30m"
+    update = "30m"
+    delete = "30m"
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  location = var.zonal_cluster ? local.zone : local.region
+  cluster  = google_container_cluster.primary.name
+
+  # if we are 'regional' i.e. in 3 zones,
+  # "1" here means "1 in each zone"
+  initial_node_count = var.dynamic_green_np_initial_node_count
+
+  autoscaling {
+    min_node_count = "0"
+    max_node_count = var.enable_spotinist ? "1" : var.zonal_cluster ? var.dynamic_green_np_initial_node_count : ceil(var.max_node_count_dynamic_green / 3)
+  }
+
+  management {
+    auto_upgrade = true
+
+    # https://cloud.google.com/kubernetes-engine/docs/how-to/node-auto-repair
+    auto_repair = true
+  }
+
+  node_config {
+
+    labels = {
+      "astronomer.io/multi-tenant" = "true"
+      "astronomer.io/dynamic-pods" = "true"
+      "astronomer.io/node-pool"    = "dynamic_green"
+    }
+
+    machine_type = var.machine_type_dynamic_green
+    disk_size_gb = var.disk_size_dynamic_green
+
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/compute",
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring",
+      "https://www.googleapis.com/auth/service.management.readonly",
+      "https://www.googleapis.com/auth/servicecontrol",
+      "https://www.googleapis.com/auth/trace.append",
+    ]
+
+    dynamic "taint" {
+      for_each = var.dynamic_green_node_pool_taints
+      content {
+        effect = taint.value.effect
+        key    = taint.value.key
+        value  = taint.value.value
+      }
+    }
+
+    # COS_CONTAINERD is required for sandbox_config to work
+    image_type = var.enable_gvisor_dynamic_green ? "COS_CONTAINERD" : "COS"
+
+    # Only include sandbox config if we are using gvisor
+    dynamic "sandbox_config" {
+      for_each = var.enable_gvisor_dynamic_green ? ["placeholder"] : []
+      content {
+        sandbox_type = "gvisor"
+      }
+    }
+
+  }
+}
+
+## Platform node-pool blue
+
 resource "google_container_node_pool" "node_pool_platform" {
 
   count = var.enable_blue_platform_node_pool ? 1 : 0
 
   provider = google-beta
+  version  = var.kube_version_gke
```

Comment:
It seems to me like these `version  = var.kube_version_gke ` additions should go away since we are using GKE Channels.

Reply:
should we deal w/ that during this merging process, or come back and fix later if we want?  i ask as we have been running terraform quite a few times this week and things are working, so i guess its just a risk vs reward thing for me....

thoughts?

Reply:
I suppose making as few changes as possible from what currently exists now before merging is a good idea.